clang/test/CAS/path-independent-cas-outputs.c

```diff
@@ -9,6 +9,25 @@
 // RUN:   %clang -target x86_64-apple-macos11 -c %s -o %t/b/t2.o -MMD -MT dependencies -MF %t/b/t2.d --serialize-diagnostics %t/b/t2.dia -Rcompile-job-cache \
 // RUN:   2>&1 | FileCheck %s --check-prefix=CACHE-HIT
 
+// Check that output path is correctly detected with -working-directory.
+
+// RUN: env LLVM_CACHE_CAS_PATH=%t/cas %clang-cache \
+// RUN:   %clang -target x86_64-apple-macos11 -c %s -working-directory %t -o a/t1_working_dir.o -DNEW_FLAG -Rcompile-job-cache \
+// RUN:   2>&1 | FileCheck %s --check-prefix=CACHE-MISS
+// RUN: env LLVM_CACHE_CAS_PATH=%t/cas %clang-cache \
+// RUN:   %clang -target x86_64-apple-macos11 -c %s -working-directory %t -o b/t2_working_dir.o -DNEW_FLAG -Rcompile-job-cache \
```

Comment:
Would this be a cache-hit if `-working-directory` is pointing to a different location?

Reply:
No, because `-working-directory` is used for input paths too, not just outputs.  You could get cache hits by using prefix mapping + -working-directory, but not just changing -working-directory.

Reply:
How about include-tree?

Reply:
Currently it is a miss since we don't do anything special for the `-working-directory` option, but I think in principle we could make it a hit. We would either have to conditionally drop working-directory from the cache key if we are using include-tree, or we would have to make all the output paths absolute and then drop -working-directory when we produce the -cc1 invocation in the first place.  I would prefer the second approach since if there are any relative paths we didn't think of they would then fail deterministically, whereas if we just remove it from the cache key we would get non-deterministic results.